src/cli_plugin/install/pack.js

```diff
@@ -0,0 +1,131 @@
+import _ from 'lodash';
+import { listFiles, extractFiles } from './zip';
+import { resolve } from 'path';
+import { sync as rimrafSync } from 'rimraf';
+import validate from 'validate-npm-package-name';
+
+//*****************************************
+//Return a list of package.json files in the archive
+//*****************************************
+async function listPackages(settings) {
+  const regExp = new RegExp('(kibana/([^/]+))/package.json', 'i');
+  const archiveFiles = await listFiles(settings.tempArchiveFile);
+
+  let packages = archiveFiles.map((file) => {
+    file = file.replace(/\\/g, '/');
+    const matches = file.match(regExp);
+    if (matches) {
+      return {
+        file: matches[0],
+        folder: matches[2]
+      };
+    }
+  });
+
+  packages = _.chain(packages).compact().uniq().value();
+
+  return packages;
+}
+
+//*****************************************
+//Extract the package.json files into the workingPath
+//*****************************************
+async function extractPackageFiles(settings, packages) {
+  const filter = {
+    files: packages.map((pkg) => pkg.file)
+  };
+  await extractFiles(settings.tempArchiveFile, settings.workingPath, 0, filter);
+}
+
+//*****************************************
+//Extract the package.json files into the workingPath
+//*****************************************
+function deletePackageFiles(settings, packages) {
+  packages.forEach((pkg) => {
+    const fullPath = resolve(settings.workingPath, 'kibana');
+    rimrafSync(fullPath);
+  });
+}
+
+//*****************************************
+//Check the plugin name
+//*****************************************
+function validatePackageName(plugin) {
```

Comment:
Validation generally implies true/false.  `assertValidPackageName` might be more appropriate here.